### DISCUSSION TITLE
Will `session.query(...)` ever be deprecated / removed?

### DISCUSSION THREAD

**Discussion by vkomarov-r7** [sqlalchemy/sqlalchemy#9619] [Usage Questions] [answered]
Hi all, we are in the process of planning for the `sqlalchemy` 2.0 migration. In reading the 2.0 Migration Guide, there is the following statement:

> Because the vast majority of an ORM application is expected to make use of Query objects as well as that the Query interface being available does not impact the new interface, the object will stay around in 2.0 but will no longer be part of documentation nor will it be supported for the most part.

It seems like this can be interpreted to mean that at some point the usage of `session.query(...)` will be deprecated or removed? Are there any plans for this? If so, is there any sort of timeline to the removal?

It is clear that we should be moving all of our queries over to the 2.0 query style. We are trying to ascertain the relative urgency of rewriting all of our queries into the new style from the old.

Thanks!

**Comment by zzzeek** [comment, marked as answer]
hi -

there are no plans to ever remove `session.query()`.   A few of its almost never-used features were removed in 2.0 to allow it to simplify a bit, but as it stands, it's self contained in one file and is not in the way of anything else.  There's billions of lines of code using `session.query()` and I still find myself using it sometimes so I dont see how it could ever be removed.

**Comment by vkomarov-r7** [comment]
Thanks! Would it be possible to make the migration guide docs a little less.. scary sounding in that case? :)

**Reply by zzzeek** [reply]
it shoudlnt say anywhere that it's being removed.  it uses the phrase "long term legacy".
